app/cluster/state/etcd.go

```diff
@@ -261,7 +262,31 @@ func (manager *ETCDManager) unmarshalWorkspace(raw []byte) workspace.ChangeEvent
 				manager.logger.Errorf("Failed to acknowledge workspace ID change for key: %s", req.AckKey)
 			}
 			return err
-		})
+		},
+		func(ctx context.Context, ackErr error) error {
```

Comment:
What if we merge those two callbacks ?

This signature: 
```go
func(ctx context.Context, ackErr error) error
```

can cover both ack and ack with error, if `ackErr is nil` we assume everything is ok

Reply:
Big fan of simpler APIs :+1: Pushed.

Reply:
Big fan of simpler APIs :+1: Pushed.

Reply:
Big fan of simpler APIs :+1: Pushed.